thesis-de/content-de.tex

```diff
@@ -3399,6 +3399,79 @@ \section{Akzeptanz Test}
 ist eine solche Datei beigefügt.
 \mr%
 
+\section{API Test}
+\label{sec:api_test}
+Während der Entwicklung des API Systems wurde bei der Implementierung einem
+neuen Algorithmus immer der zugehörige Test definiert. Dies wurde wie in
+Punkt~\ref{sec:unit_tests} bereits beschrieben, mit Komponententests
+durchgeführt. Speziell bei der API wurden die Komponententests in einem
+parallelen Ordner zu dem Source Code und in der gleichen Programmiersprache wie
+die API, also Java, verfasst. Das in dem in Punkt~\ref{sec:api_system} genannte
+Entwicklerstudio „Spring Tool Suite“ bietet dabei eine benutzerfreundliche
+Darstellung und Verwendung. Es stellt den zu einem Algorithmus zugehörigen
+Komponententest direkt im Programm bereit, sodass jeder Entwickler die
+zugehörigen Tests sofort erkennen kann und nicht danach suchen muss.
+
+Für das Ausführen der definierten Test muss kein eigene Laufzeit
+Infrastruktur bereitgestellt werden. Echte Komponententest werden normalerweise
+extrem schnell ausgeführt und steigerten durch die Integration in das genannte
+Entwicklerstudio die Produktivität der API Entwicklung um ein vielfaches. Durch
+die geringe Laufzeit konnten die Tests auch beliebig oft, in unserem Fall, nach
+jeder Änderung am Code ausgeführt werden, ohne das längere Wartezeiten
+entstehen.
+
+Die von Swagger bereitgestellten Testbibliotheken sind dabei sehr umfangreich
+und bieten zu jeder API Funktion auch die passenden Testdefinitionen. Die
+bereitgestellten Bibliotheken sind dabei von Mock Objekten abgeleitet. Mock
+Objekte sind in Testumgebungen Platzhalter für die im echten Code verwendeten
+Objektdefinitionen. Ein konkretes Beispiel dafür wäre die „Bean“ Definition
+von Swagger. Mit dieser können verschiedene Einstellungen, beziehungsweise
+verschiedene Definitionen von Objekten in insgesamt verschiedenen
```

Comment:
verschiedenen.. ich glaub das kommt 3 Mal in dem Satz vor... Mit dieser können unterschiedliche Einstellungen, .. muss der zwischen Satz dort drin stehen? "beziehungsweise verschiedene Definitionen"